rrdd/rrdd_server.ml

```diff
@@ -274,31 +274,57 @@ let forget_host_ds (ds_name : string) : unit =
     )
 
 let query_possible_dss rrdi =
+  (* We have data sources coming from different places, so we want the union of these;
+     this happens to be the union of the rrdi.rrd (live) and rrdi.rrd.rdd_dss (other).
+     This is straighforward, but those two sets of rrds have different types,
+     hence we need to coerce them into a common type (this is why we have
+     'description not available', and 'units unknown' etc.).
+     Deciding whether a ds is enabled is also not obvious. If we have a 'live'
+     ds, then it is enabled if it exists in the set rrdi.rrd. If we have an 'other'
+     ds, then it is enabled if it is also an enabled 'live' ds, otherwise it is
+     disabled.
```

Comment:
`rrdi.rrd (live)` should read as `rrdi.dss (live)` instead of `other` I would use `archived` to note it is attached to historical data. (I'm not too attached to the term, are there better ones?)

Reply:
that name is good